### PR TITLE
profile edit.

### PR DESCRIPTION
**Pull request by mpizzagalli**
# Summary
* Implemented a feature to be able to edit the user profile.

# Trello Card
https://trello.com/c/WatFwCUv/14-permitir-la-edicion-de-datos-del-usuario

# Screenshots
Profile view:
edit1
Once we clicked on edit:
edit2
Once we changed some field and submited:
edit3
Once we changed some field but we don't submit:
edit4
Once hit the cancel button:
edit5

### CODE REVIEW

File: src/app/components/profile/profile.jade [hunk 1]
```diff
@@ -1,23 +1,42 @@
...
+  .form-group
+    label
+      | {{ 'USER.CREATION' | translate }}
+    input.form-control(type='text', name='createdAt', ng-model='profileCtrl.user.createdAt', readonly , required)
+  button.btn.btn-primary(type="submit", ng-disabled="profileForm.$invalid", ng-show="profileCtrl.editOn")
+    | {{ 'SUBMIT' | translate }}
+a.btn.btn-primary(ng-hide="profileCtrl.editOn", ng-click="profileCtrl.activateEdit()")
+  | {{ 'USER.EDIT' | translate }}
+br
+a.btn.btn-danger(ng-show="profileCtrl.editOn", ng-click="profileCtrl.deactivateEdit()", style="margin-top:1%")
```
Comment (sbalay) [inline src/app/components/profile/profile.jade]: avoid inline styling. use a css file for that margin.

File: src/app/components/profile/profileController.js [hunk 2]
```diff
@@ -6,5 +6,23 @@ angular.module('app').controller('ProfileController', [
...
       }
     );
+
+    this.activateEdit = () => {
+      this.editOn = true;
+    };
+
+    this.deactivateEdit = () => {
+      this.editOn = false;
+      $state.reload();
```
Comment (sbalay) [inline src/app/components/profile/profileController.js]: Why this is needed?

File: src/app/components/profile/profile.jade [hunk 3]
```diff
@@ -1,23 +1,42 @@
...
-p
-  strong
-    | {{ 'USER.MAIL' | translate }} :
-  |  {{ profileCtrl.user.email }}
- p
-  strong
-    | {{ 'USER.CREATION' | translate }} :
-  |  {{ profileCtrl.user.createdAt | date: 'dd/MM/yyyy'}}
+form(name='profileForm', ng-submit="profileCtrl.updateUser(profileCtrl.user)", novalidate)
+  .form-group
```
Comment (sbalay) [inline src/app/components/profile/profile.jade]: You are using the same structure for all inputs. Maybe you can abstract this template to a directive
```
  .form-group
    label
      | {{ 'USER.SURNAME' | translate }}
    input.form-control(type='text', name='lastName', ng-model='profileCtrl.user.lastName', ng-readonly="!profileCtrl.editOn", required)
    p.help-block(ng-show="profileForm.lastName.$invalid && !profileForm.lastName.$pristine")
      | {{ 'USER.BLANK_SURNAME' | translate }}
```

Comment (sbalay) [inline src/app/components/profile/profile.jade]: Ask me for help if needed :)

File: src/app/components/profile/profileController.js (hunk 2 above)
Comment (mpizzagalli) [inline src/app/components/profile/profileController.js]: @sbalay so if someone hits the cancel button, we clean all dirty data that could have been modified.

File: src/app/components/profile/profile.jade (hunk 3 above)
Comment (mpizzagalli) [inline src/app/components/profile/profile.jade]: nice, i'm gonna try it!

File: src/app/components/profile/profileController.js (hunk 2 above)
Comment (sbalay) [inline src/app/components/profile/profileController.js]: Can you think of a better way of doing that?

**Comment by mpizzagalli** [comment]
@sbalay I'm stuck with the directive, the other fixes are already done, maybe i could leave the directive for another time and continue with the other card now. What do u think?

**Comment by sbalay** [comment]
:+1:

**Pull request merged**